src/creatures/players/cast/cast_viewer.cpp

```diff
@@ -0,0 +1,1533 @@
+	if (m_owner) {
+		m_owner->sendMarketEnter(depotId);
+	}
+}
+
+void CastViewer::sendUnjustifiedPoints(const uint8_t &dayProgress, const uint8_t &dayLeft, const uint8_t &weekProgress, const uint8_t &weekLeft, const uint8_t &monthProgress, const uint8_t &monthLeft, const uint8_t &skullDuration) {
+	if (m_owner) {
+		m_owner->sendUnjustifiedPoints(dayProgress, dayLeft, weekProgress, weekLeft, monthProgress, monthLeft, skullDuration);
+
+		for (const auto &it : m_viewers) {
+			it.first->sendUnjustifiedPoints(dayProgress, dayLeft, weekProgress, weekLeft, monthProgress, monthLeft, skullDuration);
+		}
+	}
+}
+
+void CastViewer::sendModalWindow(const ModalWindow &modalWindow) {
+	if (m_owner) {
+		m_owner->sendModalWindow(modalWindow);
+	}
+}
+
+void CastViewer::sendResourceBalance(Resource_t resourceType, uint64_t value) {
+	if (m_owner) {
+		m_owner->sendResourceBalance(resourceType, value);
+	}
+}
+
+void CastViewer::sendOpenWheelWindow(uint32_t ownerId) {
+	if (m_owner) {
+		m_owner->sendOpenWheelWindow(ownerId);
+	}
+}
+
+void CastViewer::sendCreatureSay(std::shared_ptr<Creature> creature, SpeakClasses type, const std::string &text, const Position* pos) {
+	if (m_owner) {
+		m_owner->sendCreatureSay(creature, type, text, pos);
+
+		if (type == TALKTYPE_PRIVATE_FROM) {
+			return;
+		}
+
+		for (const auto &it : m_viewers) {
+			it.first->sendCreatureSay(creature, type, text, pos);
+		}
+	}
+}
+
+void CastViewer::disconnectClient(const std::string &message) const {
+	if (m_owner) {
+		m_owner->disconnectClient(message);
+	}
+}
+
+void CastViewer::addViewer(ProtocolGame_ptr client, bool spy) {
+	if (m_viewers.find(client) != m_viewers.end()) {
+		return;
+	}
+
+	auto viewerId = m_viewers.size() + 1;
+	std::string guestString = fmt::format("Guest-{}", viewerId);
+
+	m_viewers[client] = std::make_pair(guestString, m_id);
+
+	if (!spy) {
+		sendChannelMessage("", fmt::format("{} has entered the cast.", guestString), TALKTYPE_CHANNEL_O, CHANNEL_CAST);
+
+		if (m_viewers.size() > m_castLiveRecord) {
+			m_castLiveRecord = m_viewers.size();
+			sendChannelMessage("", fmt::format("New record: {} people are watching your livestream now.", std::to_string(m_castLiveRecord)), TALKTYPE_CHANNEL_O, CHANNEL_CAST);
+		}
+	}
+}
+
+void CastViewer::removeViewer(ProtocolGame_ptr client, bool spy) {
+	auto it = m_viewers.find(client);
+	if (it == m_viewers.end()) {
+		return;
+	}
+
+	auto mit = std::find(m_mutes.begin(), m_mutes.end(), it->second.first);
+	if (mit != m_mutes.end()) {
+		m_mutes.erase(mit);
+	}
+
+	if (!spy) {
+		sendChannelMessage("", fmt::format("{} has left the cast.", it->second.first), TALKTYPE_CHANNEL_O, CHANNEL_CAST);
+	}
+
+	m_viewers.erase(it);
+}
+
+void CastViewer::handle(ProtocolGame_ptr client, const std::string &text, uint16_t channelId) {
+	if (!m_owner) {
+		return;
+	}
+
+	auto sit = m_viewers.find(client);
+	if (sit == m_viewers.end()) {
+		return;
+	}
+
+	const int64_t &now = OTSYS_TIME();
```

Comment:
You're right. Using std::chrono::steady_clock makes more sense here since it avoids issues related to daylight savings and system time adjustments. I’ll make the change to ensure more consistent behavior across the code.

Reply:
Going through the current uses of `OTSYS_TIME`, they all seem to be used for relative time ("schedule X in 2 seconds") rather than absolute time ("schedule Y at 14:00"). Perhaps it's cleaner to use `std::chrono::steady_clock` in `OTSYS_TIME`?